R/round.r

```diff
@@ -76,25 +77,30 @@ floor_date <- function(x, unit = c("second","minute","hour","day", "week", "mont
 #' # "2009-09-01 CDT"
 #' ceiling_date(x, "year")
 #' # "2010-01-01 CST"
-ceiling_date <- function(x, unit = c("second","minute","hour","day", "week", "month", "year")) {
-	unit <- match.arg(unit) 
-  
+ceiling_date <- function(x, unit = c("second","minute","hour","day", "week", "month", "year", "quarter")) {
+	unit <- match.arg(unit)
+
 	if(!length(x)) return(x)
-  
+
   if (unit == "second") {
     second(x) <- ceiling(second(x))
     return(x)
   }
-  
+
+  if(unit != "quarter") {
 	y <- floor_date(x - dseconds(1), unit)
-	
+  } else {
+      y <- floor_date(x, unit)
+  }
+
 	switch(unit,
 		minute = minute(y) <- minute(y) + 1,
 		hour =   hour(y) <- hour(y) + 1,
 		day =    yday(y) <- yday(y) + 1,
 		week =   week(y) <- week(y) + 1,
 		month =  month(y) <- month(y) + 1,
-		year =   year(y) <- year(y) + 1
+		year =   year(y) <- year(y) + 1,
+        quarter = { month(y) <- month(y) + 3; y <- y - ddays(1) }
```

Comment:
The issue reporter wanted ceiling to return the last day of the quarter. This line adds three months to the first day of the quarter, then subtracts one (to handle months with 28/30/31 days), instead of adding two months, and having to figure out how long a month is to get the last day.